### PR TITLE
Scale down databricks testing for the Delta Lake connector

### PR DESCRIPTION
**Pull request by findinpath**
## Description
This is a pragmatic decision driven by the high latency between the
GitHub runner and the Databricks cluster.

Keep testing on Databricks only the scenarios containing
critical functionality for the Delta Lake connector.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by electrum — APPROVED** [review]

**Comment by electrum** [comment]
/test-with-secrets sha=2bdfa9a39c07511be184e12e45e3b3d9712ef019

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeInsertCompatibility.java [hunk 1]
```diff
@@ -392,39 +386,11 @@ public void testPartitionedInsertNonLowercaseColumnsCompatibility()
...
-    }
-
     /**
      * Smoke test compression when writing to a Delta table. It's verified that writer doesn't fail and reads succeed,
      * but it's not verified that compression actually takes place.
      */
-    @Test(groups = {DELTA_LAKE_DATABRICKS, PROFILE_SPECIFIC_TESTS}, dataProvider = "compressionCodecs")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS}, dataProvider = "compressionCodecs")
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
     public void testCompression(String compressionCodec)
```
Comment (electrum) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeInsertCompatibility.java:395]: This seems to cause a failure:
```
2023-09-28 00:54:56 INFO: FAILURE     /    io.trino.tests.product.deltalake.TestDeltaLakeInsertCompatibility.testCompression [ZSTD] (Groups: profile_specific_tests, delta-lake-databricks, delta-lake-oss) took 0.9 seconds
2023-09-28 00:54:56 SEVERE: Failure cause:
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.orElseThrow(Optional.java:377)
	at io.trino.tests.product.deltalake.TestDeltaLakeInsertCompatibility.testCompression(TestDeltaLakeInsertCompatibility.java:420)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

Comment (electrum) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeInsertCompatibility.java:395]: The `databricksRuntimeVersion` code below needs to be removed

**Comment by ebyhr** [comment]
13.3 job took 55 minutes https://github.com/trinodb/trino/actions/runs/6330216462/job/17193158821. Can you confirm the longer tests and exclude them if possible?

**Comment by ebyhr** [comment]
/test-with-secrets sha=ab9d8da1addf0abfe6353f480073a36c1363dbc7

**Comment by findinpath** [comment]
databricks 13.3 job takes now 40m
https://github.com/trinodb/trino/actions/runs/6336987272/job/17212539826
delta-lake-oss job takes now 50m
https://github.com/trinodb/trino/actions/runs/6336222478/job/17211585783?pr=19180

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckConstraintCompatibility.java [hunk 2]
```diff
@@ -37,8 +33,7 @@
 public class TestDeltaLakeCheckConstraintCompatibility
         extends BaseTestDeltaLakeS3Storage
 {
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS}, dataProvider = "checkConstraints")
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS}, dataProvider = "checkConstraints")
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckConstraintCompatibility.java:36]: This class contains the following TODO comment. I think we can remove it now. 
```
// TODO: Add DELTA_LAKE_DATABRICKS groups once flakiness on Databricks is resolved
```

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java [hunk 3]
```diff
@@ -427,8 +447,7 @@ public void testDeletionVectorsAcrossAddFile(boolean partitioned)
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java:450]: Only running with DELTA_LAKE_OSS doesn't give helpful test coverage:
```java
    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
    public void testDeletionVectorsTruncateTable()
    {
        testDeletionVectorsDeleteAll(tableName -> {
            if (databricksRuntimeVersion.isPresent()) {
                onDelta().executeQuery("TRUNCATE TABLE default." + tableName);
            }
            else {
                assertThatThrownBy(() -> onDelta().executeQuery("TRUNCATE TABLE default." + tableName))
                        .hasMessageContaining("Table does not support truncates");
                throw new SkipException("OSS Delta Lake doesn't support truncating tables with deletion vector");
            }
        });
    }
```

Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java:450]: I noticed it the first time i was doing changes, but forgot yesterday about it.

**Comment by ebyhr** [comment]
/test-with-secrets sha=74b9fe9d2eaea3aa8bee04e47d6473c8f758f6a0

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckpointsCompatibility.java [hunk 4]
```diff
@@ -285,7 +283,13 @@ private void testCheckpointMinMaxStatisticsForRowType(Consumer<String> sqlExecut
...
             // Assert min/max queries can be computed from just metadata
             String explainSelectMax = getOnlyElement(onDelta().executeQuery("EXPLAIN SELECT max(root.entry_one) FROM default." + tableName).column(1));
-            String column = databricksRuntimeVersion.isAtLeast(DATABRICKS_104_RUNTIME_VERSION) ? "root.entry_one" : "root.entry_one AS `entry_one`";
+            String column;
+            if (databricksRuntimeVersion.isPresent()) {
+                column = databricksRuntimeVersion.get().isAtLeast(DATABRICKS_104_RUNTIME_VERSION) ? "root.entry_one" : "root.entry_one AS `entry_one`";
+            }
+            else {
+                column = "root.entry_one";
+            }
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckpointsCompatibility.java:292]: This `else` condition seems redundant. `testCheckpointMinMaxStatisticsForRowType` doesn't run on OSS Delta Lake. 

Same for `testCheckpointNullStatisticsForRowType`.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**